### ISSUE TITLE
construct "BANANA" returns an incorrect result

### ISSUE THREAD

**Issue by GregorySchwartz** [bos/suffixtree#4]
The result of `construct "BANANA"` returns a different result than the classic example, as shown in https://upload.wikimedia.org/wikipedia/commons/d/d2/Suffix_tree_BANANA.svg on Wikipedia. Is this expected behavior? How would this differ from the classical suffix tree?

**Comment by GregorySchwartz** [comment]
Actually, I half take it back. `construct "BANANA$"` does in fact work, but shouldn't the terminator always be taken into account?

**Comment by srhb** [comment]
I think that construct, given a non-unique terminating character, simply ends up constructing the implicit suffix tree.

**Comment by xogcox** [comment]
> I think that construct, given a non-unique terminating character, simply ends up constructing the implicit suffix tree.

Yes, the example in `examples/UniqueMatch.hs` suggests that users are expected to add a terminating character. It would probably be a good idea to either fix this so that the terminating character is added automatically, or document that it is needed.
